src/api/cluster_create.go

```diff
@@ -25,6 +25,7 @@ import (
 	"github.com/sirupsen/logrus"
 
 	"github.com/banzaicloud/pipeline/.gen/pipeline/pipeline"
+	"github.com/banzaicloud/pipeline/internal/cluster/clusteradapter"
```

Comment:
**Required**: we **MUST NOT** create this dependency chain here, I intentionally wrote the example I provided in my last comment, because I do not want this package to depend on the error type or interface directly.
Not found errors are defined in a lot of places, we only care about the interface, we should redefine the interface in src/api/error.go as I wrote in my last comment with the helper function and then we can user that.

Reply:
Sorry I did not realize your point, fixed!